File: LeetCode/python3/378.py
class Solution:
    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:
        heap = []
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                heapq.heappush(heap, matrix[i][j])
        return heapq.nsmallest(k, heap)[-1]

File: LeetCode/cpp/378.cpp
class Solution {
public:
    int kthSmallest(vector<vector<int>> &matrix, int k) {
        int n = matrix.size();
        if (n == 0)
            return 0;
        priority_queue<int> heap;
        for (auto &m: matrix) {
            for (auto &num: m) {
                if (heap.size() > k)
                    heap.pop();
                if (heap.size() < k || num < heap.top())
                    heap.push(num);
                else
                    break;
            }
        }
        return heap.top();
    }
};